src/urllib3/poolmanager.py

```diff
@@ -326,14 +325,32 @@ def _merge_pool_kwargs(self, override):
     def _proxy_requires_url_absolute_form(self, parsed_url):
         """
         Indicates if the proxy requires the complete destination URL in the
-        request.
-
-        Normally this is only needed when not using an HTTP CONNECT tunnel.
+        request.  Normally this is only needed when not using an HTTP CONNECT
+        tunnel.
         """
         if self.proxy is None:
             return False
 
-        return parsed_url.scheme == "http" or self.proxy.scheme == "https"
+        return not connection_requires_http_tunnel(
+            self.proxy, self.proxy_config, parsed_url.scheme
+        )
+
+    def _validate_proxy_scheme_url_selection(self, url_scheme):
+        """
+        Validates that were not attempting to do TLS in TLS connections on
+        Python2 or with unsupported SSL implementations.
+        """
+        if self.proxy is None or url_scheme != "https":
+            return
+
+        if self.proxy.scheme != "https":
+            return
+
+        if not PY3 and not self.proxy_config.use_forwarding_for_https:
+            raise ProxySchemeUnsupported(
+                "Contacting HTTPS destinations through HTTPS proxies is not "
+                "supported on Python 2."
```

Comment:
Maybe should add `"through HTTPS proxies 'via CONNECT tunnels' is not supported in Python 2"`. Maybe we can mention `use_forwarding_for_https` option? Just worried users will blindly tick the box but hopefully most users trust their proxies... right? :shrug: 

also nit: remove the period

Reply:
Decided not to add the mention of ``use_forwarding_for_https``. If users need the feature they should research and understand it instead of blindly enabling it.